### PR TITLE
Move materials and kernel for electrochemical phase-field model from Marmot to MOOSE

### LINKED ISSUE #19339: Move materials and kernel for electrochemical phase-field sintering model to MOOSE
## Reason
Materials needed for the electrochemical phase-field sintering model were created in Marmot. To facilitate collaboration with external users and remove dependency on Marmot from Malamute, these materials are being moved to MOOSE. No nuclear-specific information is contained in these materials.

## Design
`ElectrochemicalSinteringMaterial`, `ElectrochemicalDefectMaterial`, and the kernel `MaskedExponential` will be moved from Marmot to the MOOSE phase-field module.

## Impact
Add something new

### PR DESCRIPTION
**Pull request by laagesen**
Closes #19339

### CODE REVIEW

**Comment by dschwen** [comment]
Can you prepare a PR in Marmot that removes these objects, first?

File: modules/phase_field/test/tests/electrochem_sintering/tests
Comment (dschwen) [inline modules/phase_field/test/tests/electrochem_sintering/tests:7]:
```suggestion
[Tests]
  [ElectrochemicalSintering]
    type = 'Exodiff'
    input = 'ElectrochemicalSintering_test.i'
    exodiff = 'ElectrochemicalSintering_test_out.e'
  []
[]
```

File: modules/phase_field/test/tests/electrochem_sintering/ElectrochemicalSintering_test.i
Comment (dschwen) [inline modules/phase_field/test/tests/electrochem_sintering/ElectrochemicalSintering_test.i:506]: can you `hit format` this to use the new syntax? (like the tests file below)

**Review by dschwen — COMMENTED** [review]
Looks good. Some minor comments.

**Comment by laagesen** [comment]
Corresponding Marmot MR is ready, !540

**Comment by dschwen** [comment]
Marmot MR is merged. So hopefully the internal app tests will just pass...

**Comment by laagesen** [comment]
Looks like internal apps failed- hopefully just an issue of the changes not propagated through yet?

**Comment by dschwen** [comment]
Well, I suppose the marmot MR must make it into master first

File: modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C
Comment (jiangwen84) [inline modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C:333]: Any reason to add this option but not support it?

File: modules/phase_field/test/tests/electrochem_sintering/tests
Comment (jiangwen84) [inline modules/phase_field/test/tests/electrochem_sintering/tests:6]: I think you need to add `design`, `issue` staff, correct?

File: modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C
Comment (laagesen) [reply modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C:333]: We have a similar model that does not include electrochemical effects, that does have the ideal solution model implemented. I wanted to leave this in place as a more descriptive error in case there are some users that try to move input files from the non-electrochemical model over so they would understand better why the ideal solution option isn't available.

File: modules/phase_field/test/tests/electrochem_sintering/tests
Comment (laagesen) [reply modules/phase_field/test/tests/electrochem_sintering/tests:6]: Yes you're right, I will fix it. Actually I'm surprised the test system didn't throw an error with that not being there

File: modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C
Comment (jiangwen84) [reply modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C:333]: I got it and it makes sense.

Would it be better to move this error into the constructor? And change it to ParamError?

Comment (laagesen) [reply modules/phase_field/src/materials/ElectrochemicalSinteringMaterial.C:333]: I'd like to implement the ideal solution case eventually- I think leaving it in the switch statement here leaves the door open to do that eventually more easily

**Review by jiangwen84 — APPROVED** [review]

**Comment by loganharbour** [comment]
...no fix for Freya?

**Comment by loganharbour** [comment]
> ix for Freya?

Fix is up in freya. Just waiting for an approval from @sapitts or @laagesen.

**Comment by sapitts** [comment]
> > ...no fix for Freya?
> 
> Fix is up in freya. Just waiting for an approval from @sapitts or @laagesen.

Done

**Pull request merged**
